### DISCUSSION TITLE
Mouse Input for MRTK UI not working in built .exe

### DISCUSSION THREAD

**Discussion by halfpasttense** [microsoft/MixedRealityToolkit-Unity#11761] [Q&A] [answered]
Any UI built with MRTK UX components appears to work with mouse input only in the Unity Editor, but stops working completely when built and ran as an exe. Is this the expected behavior? Is there any way around this? I need to reuse some of the UI I've built using MRTK for a desktop windows application as well, so I need it to work with mouse input.

**Comment by AMollis** [comment]
@halfpasttense , is this a question about MRTK2 or MRTK3?   Are you running the EXE on HoloLens or just the PC?

**Reply by halfpasttense** [reply]
This is using MRTK3 and I'm running it on Windows PC as a desktop application.

**Reply by AMollis** [reply]
In MRTK3 mouse input will work in the desktop app.  Make sure you have the [UGUIInputAdapter](https://github.com/microsoft/MixedRealityToolkit-Unity/blob/mrtk3/com.microsoft.mrtk.uxcore/Interop/UGUIInputAdapter.cs) component on your interactables.  This will convert UnityUI event to XR interactions.  

I recommend trying out the "HandInteractionExamples" scene on desktop.  I've confirmed that this works on a desktop EXE with a mouse.

**Reply by AMollis** [reply]
Also...desktop is not an officially supported platform, so your milage may vary.  MRTK3 targets OpenXR headsets and is primarily tested on HoloLens 2.

**Reply by halfpasttense** [reply]
Yeah everything has the UGUIInputAdapter, so I'm not sure why it's not working. If I use any regular Unity UI components in the same project, mouse input works fine on them. It's just all the MRTK3 UI components that don't work. I'll try that example you mentioned and see if that works for me when compiled to exe.

**Reply by AMollis** [reply]
Also, to confirm, is your scene setup with the default `MRTK XR Rig` prefab?   This prefab contains all the necessary MRTK3 interactors.

**Reply by halfpasttense** [reply, marked as answer]
We figured it out after looking for differences between our project and the HandInteractionExamples demo project. What breaks mouse interaction with UI is having the "Hand Tracking" OpenXR Feature Group enabled for Windows platform.  Turning this off makes mouse interaction with UI work.
